types/graphql-upload/index.d.ts

```diff
@@ -1,53 +1,10 @@
-// Type definitions for graphql-upload 8.0
+// Type definitions for graphql-upload 15.0
 // Project: https://github.com/jaydenseric/graphql-upload#readme
 // Definitions by: Mike Marcacci <https://github.com/mike-marcacci>
+//                 Simon Feiden <https://github.com/rdsfj>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 4.1
+// TypeScript Version: 4.8
 
-/* tslint:disable:no-unnecessary-generics */
-
-import { IncomingMessage, ServerResponse } from "http";
-import { GraphQLScalarType } from "graphql";
-import { RequestHandler } from "express";
-import { DefaultContext, DefaultState, Middleware } from "koa";
-import { ReadStream } from "fs-capacitor";
-
-export interface UploadOptions {
-  maxFieldSize?: number | undefined;
-  maxFileSize?: number | undefined;
-  maxFiles?: number | undefined;
-}
-
-export interface GraphQLOperation {
-  query: string;
-  operationName?: null | string | undefined;
-  variables?: null | unknown | undefined;
-}
-
-export function processRequest(
-  request: IncomingMessage,
-  response: ServerResponse,
-  uploadOptions?: UploadOptions
-): Promise<GraphQLOperation | GraphQLOperation[]>;
-
-export function graphqlUploadExpress(
-  uploadOptions?: UploadOptions
-): RequestHandler;
-
-export function graphqlUploadKoa <StateT = DefaultState, ContextT = DefaultContext>(
-  uploadOptions?: UploadOptions
-): Middleware<StateT, ContextT>;
-
-export const GraphQLUpload: GraphQLScalarType;
-
-export interface FileUpload {
-  filename: string;
-  mimetype: string;
-  encoding: string;
-  createReadStream(): ReadStream;
-}
-
-export class Upload {
-  promise: Promise<FileUpload>;
-  file?: FileUpload;
-}
+// Linter does not accept files without definitions. Disabling the rule at
+// the top of the file doesn't work either: It breaks the meta data parser.
+type Dummy = never;
```

Comment:
Because there are no imports or exports in this file, this type is a global, visible to everyone in their completion lists if they have these types installed 😅. Does `export {}` not satisfy the linter?